File: data/multilang_comparison/program_go/src/func12.go
package main
import "math/rand"
var _ = rand.Intn
func Func12(vars *ArrayParam, PATH0 uint64 , loopsFactor int) *Array{
   var pCounter int = vars.Size;
   _ = pCounter
   if (PATH0 & 1) != 0 {
      var array66 *Array
      if pCounter > 0 {
          array66 = vars.Data[pCounter-1]
          pCounter--
          array66.RefC++
      } else {
          array66 = &Array{
              Size: 786,
              RefC: 1,
              Id: 66,
              Data: make([]uint32, 786),
          }
      }
      var loop21 int = 0;
      var loopLimit21 int = (50)/3 + 1;
      for ; loop21 < loopLimit21; loop21++ {
         for i := 0; i < array66.Size; i++ {
             array66.Data[i]--
         }
         for i := 0; i < array66.Size; i++ {
             if array66.Data[i] == 42 {
                 return array66
             }
         }
      }
      var params0 ArrayParam
      params0.Size = 1
      params0.Data = make([]*Array, params0.Size)
      params0.Data[0] = array66
      var array67 *Array = Func24(&params0, getPath(), loopsFactor);
      params0.Data = nil
      params0.Size = 0
      array67.RefC--
      if array67.RefC == 0 {
          array67.Data = nil
          array67 = nil
      }
      array66.RefC--
      if array66.RefC == 0 {
          array66.Data = nil
          array66 = nil
      }
   
   } else {
      var params0 ArrayParam
      params0.Size = 0
      params0.Data = make([]*Array, params0.Size)
      var array90 *Array = Func16(&params0, getPath(), loopsFactor);
      params0.Data = nil
      params0.Size = 0
      array90.RefC--
      if array90.RefC == 0 {
          array90.Data = nil
          array90 = nil
      }
   }
   var array100 *Array
   if pCounter > 0 {
       array100 = vars.Data[pCounter-1]
       pCounter--
       array100.RefC++
   } else {
       array100 = &Array{
           Size: 940,
           RefC: 1,
           Id: 100,
           Data: make([]uint32, 940),
       }
   }
   return array100;
}

File: data/multilang_comparison/program_cpp_array/src/func12.cpp
#include "program_cpp_array.hpp" 
Array* func12(Array_param* vars, const unsigned long PATH0, int loopsFactor) {
   size_t pCounter = vars->size;
   if(PATH0 & 1) {
      Array* array66;
      if (pCounter > 0) {
         array66 = vars->data[--pCounter];
         array66->refC++;
         DEBUG_COPY(array66->id);
      } else {
         array66 = new Array();
         array66->size = 786;
         array66->refC = 1;
         array66->id = 66;
         array66->data = new unsigned int[array66->size]();
         DEBUG_NEW(array66->id);
      }
      unsigned int loop21 = 0;
      unsigned int loopLimit21 = (50)/3 + 1;
      for(; loop21 < loopLimit21; loop21++) {
         for (int i = 0; i < array66->size; i++) {
             array66->data[i]--;
         }
         for (int i = 0; i < array66->size; i++) {
             if (array66->data[i] == 42) {
                 return array66;
             }
         }
      }
      Array_param params0;
      params0.size = 1;
      Array* data_params0[1];
      params0.data = data_params0;
      params0.data[0] = array66;
      Array* array67 = func24(&params0, get_path(), loopsFactor);
      DEBUG_RETURN(array67->id);
      if(params0.size > 0) {
      	 params0.data.clear();
      }
      array67->refC--;
      if(array67->refC == 0) {
         delete[] array67->data;
         array67->data = nullptr;
         DEBUG_FREE(array67->id);
         delete array67;
      }
      array66->refC--;
      if(array66->refC == 0) {
         delete[] array66->data;
         array66->data = nullptr;
         DEBUG_FREE(array66->id);
         delete array66;
      }
   }
   else {
      Array_param params0;
      params0.size = 0;
      Array* data_params0[0];
      params0.data = data_params0;
      Array* array90 = func16(&params0, get_path(), loopsFactor);
      DEBUG_RETURN(array90->id);
      if(params0.size > 0) {
      	 params0.data.clear();
      }
      array90->refC--;
      if(array90->refC == 0) {
         delete[] array90->data;
         array90->data = nullptr;
         DEBUG_FREE(array90->id);
         delete array90;
      }
   }
   Array* array100;
   if (pCounter > 0) {
      array100 = vars->data[--pCounter];
      array100->refC++;
      DEBUG_COPY(array100->id);
   } else {
      array100 = new Array();
      array100->size = 940;
      array100->refC = 1;
      array100->id = 100;
      array100->data = new unsigned int[array100->size]();
      DEBUG_NEW(array100->id);
   }
   return array100;
}

